utilities/write_batch_with_index/write_batch_with_index_internal.cc

```diff
@@ -838,6 +838,9 @@ WBWIIteratorImpl::Result WriteBatchWithIndexInternal::GetFromBatchImpl(
         Traits::ClearOutput(output);
         result = WBWIIteratorImpl::Result::kError;
       }
+    } else {
+      Traits::ClearOutput(output);
+      *s = Status::OK();
```

Comment:
I am wondering if we can set the status `Status::NotFound()` here instead and return status as is when result is `kFound`, `kError`, or `kDeleted` above. (line 850 in utilities/write_batch_with_index/write_batch_with_index.cc)

Reply:
Yeah, currently the contract here is that the "real" result is in the return value, and the status output parameter comes into play with errors only. This feels like the right call considering that `WBIIteratorImpl::Result` doesn't have a 1:1 mapping with status codes: we have to differentiate between "key deleted" and "no entries for the key at all", and different clients actually handle these two cases differently (depending on whether it's a "read from batch only" or "read from batch and DB" type API).